eventuate-messaging-kafka-common/src/test/java/io/eventuate/messaging/kafka/common/EventuateKafkaMultiMessageConverterTest.java

```diff
@@ -11,152 +11,121 @@
-    System.out.println(String.format("average = %s", average));
-    System.out.println(String.format("average without min and max = %s", averageWithoutMinAndMax));
-  }
-
-  @Test
-  public void testMessageConverterRegularMessages() {
-    testMessageConverter(MESSAGES);
-  }
+  private byte[] serializedMessages;
+  private int estimatedSize;
 
   @Test
-  public void testMessageConverterNullMessages() {
-    testMessageConverter(NULL_MESSAGES, EMPTY_MESSAGES);
+  public void testMessageBuilderSimpleMessages() {
+    testMessageBuilder(new EventuateKafkaMultiMessages(SIMPLE_MESSAGES));
   }
 
   @Test
-  public void testMessageConverterEmptyMessages() {
-    testMessageConverter(EMPTY_MESSAGES, EMPTY_MESSAGES);
+  public void testMessageBuilder2ByteCharacterMessages() {
+    testMessageBuilder(new EventuateKafkaMultiMessages(TWO_BYTE_CHARACTER_MESSAGES));
+    Assert.assertEquals(serializedMessages.length, estimatedSize);
   }
 
   @Test
-  public void testMessageBuilderRegularMessages() {
-    testMessageBuilder(MESSAGES);
+  public void testMessageBuilderMessagesWithHeaders() {
+    testMessageBuilder(new EventuateKafkaMultiMessages(HEADERS, MESSAGES_WITH_HEADERS));
   }
 
   @Test
   public void testMessageBuilderNullMessages() {
-    testMessageBuilder(NULL_MESSAGES, EMPTY_MESSAGES);
+    testMessageBuilder(new EventuateKafkaMultiMessages(NULL_MESSAGES), new EventuateKafkaMultiMessages(EMPTY_MESSAGES));
   }
 
   @Test
   public void testMessageBuilderEmptyMessages() {
-    testMessageBuilder(EMPTY_MESSAGES, EMPTY_MESSAGES);
+    testMessageBuilder(new EventuateKafkaMultiMessages(EMPTY_MESSAGES), new EventuateKafkaMultiMessages(EMPTY_MESSAGES));
   }
 
   @Test
   public void testMessageBuilderSizeCheck() {
-    int sizeOfHeaderAndFirstMessage = MessageHeaderEncoder.ENCODED_LENGTH + MultiMessageEncoder.MessagesEncoder.HEADER_SIZE
-            + 2 * 4
-            + MESSAGES.get(0).getKey().length() * 2
-            + MESSAGES.get(0).getValue().length() * 2;
+    int sizeOfHeaderAndFirstMessage = MessageHeaderEncoder.ENCODED_LENGTH + MultiMessageEncoder.MessagesEncoder.HEADER_SIZE + MultiMessageEncoder.HeadersEncoder.HEADER_SIZE
+            + SIMPLE_MESSAGES.get(0).estimateSize();
 
     EventuateKafkaMultiMessageConverter.MessageBuilder messageBuilder =
             new EventuateKafkaMultiMessageConverter.MessageBuilder(sizeOfHeaderAndFirstMessage);
 
-    Assert.assertTrue(messageBuilder.addMessage(MESSAGES.get(0)));
-    Assert.assertFalse(messageBuilder.addMessage(MESSAGES.get(1)));
+    Assert.assertTrue(messageBuilder.addMessage(SIMPLE_MESSAGES.get(0)));
+    Assert.assertFalse(messageBuilder.addMessage(SIMPLE_MESSAGES.get(1)));
   }
 
   @Test
   public void testMessageBuilderHeaderSizeCheck() {
     int sizeOfFirstMessage = 2 * 4
-            + MESSAGES.get(0).getKey().length() * 2
-            + MESSAGES.get(0).getValue().length() * 2;
+            + SIMPLE_MESSAGES.get(0).getKey().length() * 2
```

Comment:
The code would be simpler if there was a MESSAGES_0_KEY, MESSAGES_1_VALUE, ...?

Reply:
ok, will do